### PR TITLE
check signed import

### PR DESCRIPTION
**Pull request by f-lukas**
https://github.com/Pandora-IsoMemo/TraceR/issues/21

### CODE REVIEW

**Comment by f-lukas** [comment]
@arunge you can start with the review. I have to check with @jan-abel-inwt if my adjustments in the docker and jenkinsfile work. Also the pipeline is failing at the moment because the secret files are not yet uploaded to jenkins.

If you want to test it locally you can create your own key pair with your script here https://github.com/Pandora-IsoMemo/TraceR/pull/10/files. You have to place the files in `inst/app`. To test how the functionalitiy works for the deployed version, you can create an `.Renviron` file with SHINYPROXY_USERID=123.

File: R/02_importModule.R [hunk 1]
```diff
@@ -76,3 +81,55 @@ importModuleServer <- function(id) {
...
+#' @export
+check_signature_validity <- function(graph_list, public_key){
+  if (is.na(Sys.getenv("SHINYPROXY_USERID", unset = NA))){
+    message <- "Found a signature in the uploaded file. Validating signatures of uploaded files is not possible for apps running locally.<br><br>Please use the <a href='https://pandoraapp.earth/app/tracer' target='_blank'>online version of the app</a> for validating files."
+    shinyalert(
+      title = "Warning",
+      text = message,
+      type = "warning",
+      html = T
+    )
```
Comment (arunge) [inline R/02_importModule.R:99]:
```suggestion
    )
  }
```

File: R/02_importModule.R [hunk 2]
```diff
@@ -76,3 +81,55 @@ importModuleServer <- function(id) {
...
+      message <- "Signature of uploaded json could not be verified."
+      type <- "warning"
+    }
+    shinyalert(
+      title = title,
+      text = message,
+      type = type
+    )
+  } else {
+    message <- "No signature was found in the uploaded json. File could not be verified."
```
Comment (arunge) [inline R/02_importModule.R:127]:
```suggestion
    message <- "No signature was found in the uploaded json. Continuing without verification."
```

File: R/02_importModule.R [hunk 3]
```diff
@@ -76,3 +81,55 @@ importModuleServer <- function(id) {
...
+      title <- "Verification failed!"
+      message <- "Signature of uploaded json could not be verified."
+      type <- "warning"
+    }
+    shinyalert(
+      title = title,
+      text = message,
+      type = type
+    )
+  } else {
```
Comment (arunge) [inline R/02_importModule.R:126]:
```suggestion
  } else {
  # Skip check of validity, because no signature
```

**Review by arunge — CHANGES_REQUESTED** [review]
I find the if-else clauses  regarding the validation is quite nested, but I do not have a better idea right now. I just added some more comments. I fine-tuned some warnings.

In general, all was working fine! :+1:  Only the case of unsigned files locally, see below.

**Comment by f-lukas** [comment]
Good point. I missed this case and applied your suggestions.

**Review by arunge — APPROVED** [review]

**Pull request merged**